yt_astro_analysis/halo_analysis/tests/test_halo_finders.py

```diff
@@ -1,61 +1,119 @@
 import os
-import shutil
 import sys
-import tempfile
+
+import pytest
 
 from yt.frontends.halo_catalog.data_structures import YTHaloCatalogDataset
 from yt.frontends.rockstar.data_structures import RockstarDataset
 from yt.loaders import load
-from yt.utilities.answer_testing.framework import FieldValuesTest, requires_ds
+from yt.testing import assert_allclose_units, requires_file
+
+_expected = {
+    "fof": {
+        ("halos", "particle_position_x"): (
+            [0.5395228362545148, 0.009322612122839679, 0.9851068812956318],
+            "unitary",
+        ),
+        ("halos", "particle_position_y"): (
+            [0.5622196217878699, 0.003041142049505119, 0.9959844650128505],
+            "unitary",
+        ),
+        ("halos", "particle_position_z"): (
+            [0.5203774727457908, 0.0007079871575761398, 0.9990256781592326],
+            "unitary",
+        ),
+        ("halos", "particle_mass"): (
+            [5874503393478.599, 719586865542.6366, 46593249543885.67],
+            "Msun",
+        ),
+    },
+    "hop": {
+        ("halos", "particle_position_x"): (
+            [0.5541340411242247, 0.054537564738846515, 0.9874987118674894],
+            "unitary",
+        ),
+        ("halos", "particle_position_y"): (
+            [0.5230348494537058, 0.02430675276218758, 0.9654696366375637],
+            "unitary",
+        ),
+        ("halos", "particle_position_z"): (
+            [0.4661933399047146, 0.0049056915980563165, 0.9985715364094563],
+            "unitary",
+        ),
+        ("halos", "particle_mass"): (
+            [13744109131864.344, 4407469551448.649, 38677794022916.664],
+            "Msun",
+        ),
+    },
+    "rockstar": {
+        ("halos", "particle_position_x"): (
+            [16.706823560058094, 1.2783197164535522, 31.622468948364258],
+            "Mpccm/h",
```

Comment:
units are pretty funky with this method. I think that's hinting to a bug, but for this PR I'm purely focused on migrating existing tests (bug-for-bug).

Reply:
@cphyc pointed out that `h` is dimensionless, so the dimensions of the result doesn't actually change, which means I don't have a problem with this after all !

Reply:
Rockstar specifically wants comoving Mpc/h as its unit, so this is ok.